server.js

```diff
@@ -47,23 +47,136 @@ app.get('/api', function apiIndex(req, res) {
+
+      {method: "GET", path: "/api/profile/projects", decsription: "Describes all my projects."}, // done
+      {method: "GET", path: "/api/profile/:profile_id/projects/:_id",description: "displays particular project info"}, //done
+      {method: "POST",path: "/api/profile/:profile_id/projects",description:"creates new project"}, //done
+      {method: "DELETE", path:"/api/profile/:profile_id/projects/:_id", description:"deletes that project"},//done
+      {method: "PUT", path:"/api/profile/:profile_id/projects/:_id", description:"Updates that project by ID"},//done
+
+      {method: "GET", path: "/api/profile/:profile_id/vacation",description:"destinations i've visited"},//done
+      {method: "POST",path:"/api/profile/:profile_id/vacation",description:"Creates an entry for vacations under planning"}
+
     ]
   })
 });
 
+/*End Point 2 - Get Profile Info*/
+app.get('/api/profile', function show(req, res) {
+  console.log("inside get profile",req.body);
+  db.Profile.find({},function(err,new_profile){
+    if(err) { console.log(" new profile not exist", err)}
+    res.json(new_profile);
+  })
+});
+
+/* End point 3 - Get All Projects */
+app.get('/api/profile/projects', function(req,res){
+  console.log("inside get projects", req);
+  db.Profile.find({},function(err,projects){
+    if(err){ console.log("Get All projects Method Error",err);}
+    res.json(projects);
+  })
+});
+
+/* End Point 4 - Create new project*/
+app.post('/api/profile/:profile_id/projects',function(req,res){
+  console.log(`inside create projects method ${req}`);
+  console.log(JSON.stringify(req.body));
+  db.Profile.findById(req.params.profile_id,function(err,foundProfile){
+
+  var newProject=new db.Projects(req.body);
+
+  foundProfile.projects.push(newProject);
+  foundProfile.save(function(err,savedprofile){
+    res.json(savedprofile);
+  })
+})
+});
+
+/* End Point 5 - Show one project by ID*/
+app.get('/api/profile/:profile_id/projects/:_id',function(req,res){
+  console.log("inside getbyID");
+  db.Profile.findById(req.params._id, function(err, foundproj) {
+    res.json(foundproj);
+  });
+
+})
+
+/*End Point 6 - delete project by ID*/
+app.delete("/api/profile/:profile_id/projects/:_id",function destroy(req, res) {
+  db.Profile.findById(req.params.profile_id, function(err, foundproj) {
+    console.log(foundproj);
+
+    var correctProj = foundproj.projects.id(req.params._id);
+    if (correctProj) {
+      correctProj.remove();
+
+      foundproj.save(function(err, saved) {
+        console.log('REMOVED ', correctProj.name);
+        res.json(correctProj);
+      });
+    }
+  });
+});
+
+/* End Point 7 - update project*/
+app.put("/api/profile/:profile_id/projects/:_id",function update(req, res) {
+  db.Profile.findById(req.params.profile_id, function(err, foundProfile) {
+    var correctProj = foundProfile.projects.id(req.params._id);
+
+    if (correctProj) {
+
+      correctProj.name = req.body.name;
```

Comment:
i like how organized your code is with brackets / parentheses aligned properly ( i have trouble with that)